src/library-authoring/create-library/hooks.js

```diff
@@ -0,0 +1,81 @@
+// @ts-check
+import { useState } from 'react';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { useFormik } from 'formik';
+import { useNavigate } from 'react-router-dom';
+import * as Yup from 'yup';
+
+import { useOrganizationListData } from '../../generic/data/apiHooks';
+import { useCreateLibraryV2 } from './data/apiHooks';
+import messages from './messages';
+
+// eslint-disable-next-line import/prefer-default-export
+export const useCreateLibraryForm = (initialValues) => {
+  const intl = useIntl();
+  const navigate = useNavigate();
+
+  const [apiError, setApiError] = useState(null);
+
+  const {
+    mutateAsync,
+  } = useCreateLibraryV2();
+
+  const specialCharsRule = /^[a-zA-Z0-9_\-.'*~\s]+$/;
+  const noSpaceRule = /^\S*$/;
+  const validSlugIdRegex = /^[a-zA-Z\d]+(?:[\w -]*[a-zA-Z\d]+)*$/;
```

Comment:
The `specialCharsRule` and `noSpaceRule` regexes are defined elsewhere in this MFE.. could these be moved out somewhere global so they can be shared?